File: Static/Identifiable/Identifiable.cpp
#include "Identifiable.hpp"

Identifiable::Identifiable() {
    m_id += 1;
    m_uniqueId = m_id;
}

Identifiable::~Identifiable() {}

int Identifiable::getUniqueId() const {
    return m_uniqueId;
}

int Identifiable::m_id = 100;

File: PHP/Static/Identifiable/Identifiable.php
<?php
    class Identifiable {
        protected $uniqueId;
        private static $id = 0;
        
        public function __construct() {
            self::$id += 1;
            $this->uniqueId = self::$id;
        }
        
        public function getUniqueId() {
            return $this->uniqueId;
        }
    }
?>
